### DISCUSSION TITLE
Mill can not find ammonite when running the repl

### DISCUSSION THREAD

**Discussion by lucc** [com-lihaoyi/mill#1396] [Q&A]
I am using mill to build and run my scala project. Now I wanted to start a repl but am confronted with a 404 HTTP error (I suppose). As I have not specified any version constraints on ammonite manually I was hoping that mill would magically pick a version that existed. Am I doing something wrong?

For reference here is my build.sc: https://github.com/lucc/ulang-tiny/blob/develop/build.sc
```
$ mill -i ulang.repl     
[30/34] ulang.resolvedAmmoniteReplIvyDeps | Downloading [2/2] artifacts (~0/0 bytes)
1 targets failed
ulang.resolvedAmmoniteReplIvyDeps 
Resolution failed for 1 modules:
--------------------------------------------
  com.lihaoyi:ammonite_2.12.10:2.3.8-46-37b110d0 
	not found: /home/luc/.ivy2/local/com.lihaoyi/ammonite_2.12.10/2.3.8-46-37b110d0/ivys/ivy.xml
	not found: https://repo1.maven.org/maven2/com/lihaoyi/ammonite_2.12.10/2.3.8-46-37b110d0/ammonite_2.12.10-2.3.8-46-37b110d0.pom
```

**Comment by lefou** [comment]
Mill will by default pick the same ammonite version which it's using internally. But as ammonite releases need to match the full scala version, and the pre-selected ammonite version (`2.3.8-46-37b110d0`) wasn't released for Scala 2.12.10, you need to specify another ammonite version by overriding `def ammoniteVersion`.

```scala
object ulang extends ScalaModule {
  override def scalaVersion = "2.12.10"
  override def ammoniteVersion = "2.4.0"
  // ...
}
```

**Reply by lefou** [reply]
Please read the comments of [issue 1438](https://github.com/com-lihaoyi/mill/issues/1438) for details.
